airflow/providers/google/cloud/transfers/gcs_to_bigquery.py

```diff
@@ -276,7 +276,7 @@ def execute(self, context):
                 encryption_configuration=self.encryption_configuration
             )
         else:
-            cursor.run_load(
+            bq_hook.insert_job(configuration=dict(
```

Comment:
That won't do:
```
airflow.exceptions.AirflowException: Unknown job type. Supported types: dict_keys(['load', 'copy', 'extract', 'query'])
```
as per docstring:
```
The configuration parameter maps directly to
            BigQuery's configuration field in the job object. See
            https://cloud.google.com/bigquery/docs/reference/v2/jobs for
            details.
```

So what I would suggest is:
- add `configuration` parameter to constructor
- if this parameter is passed then use `insert_job` method
- if not then show a deprecation warning that `Users should pass load job definition` and use the existing logic

Probably we should validate that only `load` job is passed... WDYT @edejong ?